packages/components/src/Table/Filters/RegexpFilter.js

```diff
@@ -0,0 +1,56 @@
+import Filter from './Filter';
+
+function isString(regexp) {
+	return typeof regexp === 'string';
+}
+
+function lowerCaseMatch(value, regexp) {
+	return value.toLowerCase().match(regexp);
+}
+
+/**
+ * This is used to filter string values based on regular expression.
+ * Default behaviour is: if a string value matches the regexp, then it is not filtered.
+ */
+export default class RegexpFilter extends Filter {
```

Comment:
Please avoid using class hierarchies, use composition, a class for a filter is total overkill, and local state make it hard to test and maintain.